zwrap/sample.go

```diff
@@ -98,70 +98,66 @@ func (s *sampler) With(fields ...zap.Field) zap.Logger {
 }
 
 func (s *sampler) Check(lvl zap.Level, msg string) *zap.CheckedMessage {
-	if !s.check(lvl, msg) {
+	cm := s.Logger.Check(lvl, msg)
+	if cm == nil {
 		return nil
 	}
-	return zap.NewCheckedMessage(s.Logger, lvl, msg)
+	switch lvl {
+	case zap.PanicLevel, zap.FatalLevel:
+		break
```

Comment:
Can we add a note to the `Sample` doc comment pointing out that `Panic` and `Fatal` level logs are never sampled? This isn't a change in behavior, but I neglected to document it properly.